cpp/src/arrow/compute/kernels/scalar_nested_test.cc

```diff
@@ -116,6 +117,100 @@ TEST(TestScalarNested, ListElementInvalid) {
               Raises(StatusCode::Invalid));
 }
 
+void CheckListSlice(std::shared_ptr<Array> input, std::shared_ptr<Array> expected,
+                    const ListSliceOptions& args) {
+  ASSERT_OK_AND_ASSIGN(auto result, CallFunction("list_slice", {input}, &args));
+  ASSERT_EQ(result, expected);
+}
+
+TEST(TestScalarNested, ListSliceVariableOutput) {
```

Comment:
Can you also add a case with top-level nulls? (so not only with nulls inside the list elements)

Reply:
Good point, added!